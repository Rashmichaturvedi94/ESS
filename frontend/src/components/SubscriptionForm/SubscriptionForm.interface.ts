import { FormProps } from "../../hooks/useForm.interface";

export interface SubFormValues {
  title: string;
  description: string;
  price: string;
  method: string;
}

export type SubscriptionFormProps = FormProps<SubFormValues>;
