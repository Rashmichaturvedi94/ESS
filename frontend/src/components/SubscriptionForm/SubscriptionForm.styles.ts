import styled from "styled-components";
import { InputLabel, Typography } from "@material-ui/core";

export const PriceContainer = styled.div`
  margin-top: 50px;
  flex: 2;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
`;

export const ButtonContainer = styled.div`
  margin-top: 10px;
  flex: 2;
  display: flex;
  flex-direction: row;
  justify-content: left;
  font-size: 40px;
`;

export const TitleLabel = styled(Typography)`
  font-size: 40px;
  font-weight: 700;
  margin-top: 50px;
`;
export const DescLabel = styled(InputLabel)`
  font-size: 25px;
`;

export const PaymentSelector = styled(InputLabel)`
  font-size: 40px;
  margin-top: 40px;
`;

export const PriceLabel = styled(InputLabel)`
  font-size: 25px;
  margin-top: 10px;
  margin-left: 20px;
`;
