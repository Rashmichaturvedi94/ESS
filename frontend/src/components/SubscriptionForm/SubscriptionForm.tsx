import React, { FC } from "react";
import { Box, Button, FormHelperText } from "@material-ui/core";
import CreditCardIcon from "@material-ui/icons/CreditCard";

import {
  SubFormValues,
  SubscriptionFormProps,
} from "./SubscriptionForm.interface";
import {
  TitleLabel,
  DescLabel,
  PriceContainer,
  PaymentSelector,
  ButtonContainer,
  PriceLabel,
} from "./SubscriptionForm.styles";
import { useForm } from "../../hooks/useForm";

const initialValues: SubFormValues = {
  title: "test",
  description: "test",
  price: 0,
  course: 0,
  subscriber: 0,
  active: false,
};

export const SubscriptionForm: FC<SubscriptionFormProps> = (props) => {
  const { errors, values, handleSubmit } = useForm({
    initialValues,
    ...props,
  });

  return (
    <form onSubmit={handleSubmit}>
      <Box display="flex" flexDirection="column" gridRowGap={20}>
        {errors?.detail && (
          <FormHelperText error>{errors?.detail}</FormHelperText>
        )}

        <TitleLabel>{values.title}</TitleLabel>
        <DescLabel>{values.description}</DescLabel>
        <PriceContainer>
          <CreditCardIcon style={{ fontSize: 40 }} />
          <PriceLabel>Price:</PriceLabel>
          <PriceLabel>${values.price}</PriceLabel>
        </PriceContainer>
        <PaymentSelector>Choose Payment method</PaymentSelector>
        <ButtonContainer>
          <Button
            type="submit"
            variant="contained"
            style={{ background: "grey", marginLeft: 10 }}
          >
            Pay by Card
          </Button>
          <Button
            type="submit"
            variant="contained"
            style={{ background: "grey", marginLeft: 50 }}
          >
            Pay by Blik
          </Button>
        </ButtonContainer>
      </Box>
    </form>
  );
};
