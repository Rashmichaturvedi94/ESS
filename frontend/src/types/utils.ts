export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
