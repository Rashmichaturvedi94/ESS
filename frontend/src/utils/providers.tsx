import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
