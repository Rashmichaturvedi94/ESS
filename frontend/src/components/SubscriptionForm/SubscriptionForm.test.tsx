import React from 'react';
import { render } from '../../utils/testUtils';
import { SubscriptionForm } from './SubscriptionForm';

describe('components/SubscriptionForm', () => {
  it('should render', () => {
    const { container } = render(<SubscriptionForm />);
    expect(container).toMatchSnapshot();
  });
});
