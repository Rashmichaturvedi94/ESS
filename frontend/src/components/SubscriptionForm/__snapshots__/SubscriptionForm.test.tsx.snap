// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SubscriptionForm should render 1`] = `
<div>
  <form>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <p
        class="MuiTypography-root sc-dlnjwi gIeIxe MuiTypography-body1"
      >
        test
      </p>
      <label
        class="MuiFormLabel-root MuiInputLabel-root sc-hKFxyN fIcukl MuiInputLabel-shrink"
        data-shrink="true"
      >
        test
      </label>
      <div
        class="sc-bdnxRM ijokhQ"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="font-size: 40px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
          />
        </svg>
        <label
          class="MuiFormLabel-root MuiInputLabel-root sc-jSFjdj jfMEkW MuiInputLabel-shrink"
          data-shrink="true"
        >
          Price:
        </label>
        <label
          class="MuiFormLabel-root MuiInputLabel-root sc-jSFjdj jfMEkW MuiInputLabel-shrink"
          data-shrink="true"
        >
          $
          0
        </label>
      </div>
      <label
        class="MuiFormLabel-root MuiInputLabel-root sc-eCApnc iNiMum MuiInputLabel-shrink"
        data-shrink="true"
      >
        Choose Payment method
      </label>
      <div
        class="sc-gtsrHT cMLYIe"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          style="background: grey; margin-left: 10px;"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Pay by Card
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          style="background: grey; margin-left: 50px;"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Pay by Blik
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
